sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/FlatteningDeserializer.java

```diff
@@ -137,7 +137,9 @@ public Object deserialize(JsonParser jp, DeserializationContext cxt) throws IOEx
 
     @Override
     public void resolve(DeserializationContext cxt) throws JsonMappingException {
-        ((ResolvableDeserializer) this.defaultDeserializer).resolve(cxt);
+        if (this.defaultDeserializer instanceof ResolvableDeserializer) {
```

Comment:
@anuchandy do you know what this method is doing? Wanted to ensure that this change was safe.